lib/streaming-pipeline.js

```diff
@@ -42,14 +44,46 @@ module.exports = class StreamingPipeline extends Writable {
             logger('Ending input streams');
             this._endInputs();
         });
-        this.on('error', () => {
-            logger('Error occurred - ending pipeline now');
-            this.end();
+        this.on('error', (err) => {
+            logger('Error occurred - canceling invocation now');
+            this._cancelCall(grpcStream, err);
         });
         this._invokeHooks();
         logger('Streaming pipeline initialized');
     }
 
+    _cancelCall(grpcStream, err) {
+        switch (err.type) {
+            case 'error-streaming-invalid-input-signal':
+            case 'error-streaming-invalid-output-count':
+            case 'error-streaming-missing-start-signal':
+            case 'error-streaming-too-many-start-signals':
+                grpcStream.call.cancelWithStatus(
+                    grpc.status.UNKNOWN,
+                    `Invoker: Protocol Violation: ${err.cause}`
+                );
+                break;
+            case 'error-output-content-type-unsupported':
+                grpcStream.call.cancelWithStatus(
+                    grpc.status.INVALID_ARGUMENT,
+                    `Invoker: Not Acceptable: ${err.cause}`
```

Comment:
Would be nice to extract the prefix, which is public API, in a `const`

Reply:
done